source/develop/release-management/features/virt/enhance-import-export-with-ova.md

```diff
@@ -0,0 +1,135 @@
+8. Imageio-proxy sends the engine a URL to write to. This URL is sent to the front-end.
+9. The front-end streams the **whole OVA** to imageio-proxy, that streams it to imageio.
+10. Imageio reads the data and according to the import parameters streams the data into the corresponding target disks.
+
+Notes:
+
+1. The described process assumes that the disks are not compressed within the OVA. In order to support this, the back-end would need to pass an indication to imageio-proxy of whether or not each disk is compressed (step 7) so the proxy could uncompress the data on-the-fly while streaming it to the destination volume (step 10).
+2. In the figure above the format of the disks within the OVA is QCOW that is going to be the default format in OVA files generated by oVirt. However, we should support import of OVA that contains RAW disks as well.
+3. In the figure above the destination volumes are of type RAW. While this is preferable because these volumes would serve as base volumes and the performance is better by having them of type RAW, the conversion of volumes on-the-fly from QCOW to RAW may be postponed and so at the beginning the target volumes would be of type QCOW.
+
+### Uploading an OVA Generated by an External Source
+The following figure depicts the import process of an OVA that was generated by external source:
+
+![](../../../../images/wiki/upload-external-ova.png)  
+
+The process differ from import of OVA generated by oVirt in the following steps:  
+6. Only one temporary disk is created on the storage domain.  
+10. The whole OVA is streamed into the created disk.  
+11. Additional step that converts the streamed OVA and removed the temporary disk.  
+
+Note:  
+
+1. This process if not efficient in time and space, due to the creation of the temporary disk. Thus, we will suggest to make the OVA accessible to the host instead, especially in the case of mass import of OVA files. However, this process can be handy for a one upload with loose time- and space- constraints.  
+2. Therefore, it is a second-priority task.
+
+## Exporting a VM or Template as an OVA
+The user will be able to specify a VM (or a template) to be exported as an OVA. In that case, the user needs to provide a path on a host where the OVA file should be created.  
+
+The following figure depicts the process of exporting a VM with a single disk as OVA:  
+
+![](../../../../images/wiki/export-ova.png)  
+
+1. The front-end sends the back-end a VM and a path on a host.
+2. The back-end asks imagio-proxy to initialize a file at the given path with an estimated size of the produced OVA.
+3. Imageio creates the OVA file.
+4. The back-end sends imageio-proxy the disk to stream, which sends it to imageio.
+5. Imageio reads the disk and streams it into a QCOW volume within the OVA.
+6. The back-end sends the complete OVF specification to imageio-proxy, which sends it to imageio that streams it into the OVA.
```

Comment:
"The complete OVF specification is sent from the back-end to imageio-proxy. imageio-proxy sends the specification to imageio, and imageio streams it into the OVA file.